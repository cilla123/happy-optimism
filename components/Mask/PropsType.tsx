export default interface PropsType {
  visible?: boolean;
  type?: 'transparent' | 'normal';
  onClose?: () => void;
  style?: React.CSSProperties;
}
