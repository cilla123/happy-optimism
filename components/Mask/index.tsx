import React, { PureComponent } from 'react';
import classnames from 'classnames';
import PropsType from './PropsType';

export interface MaskProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Mask extends PureComponent<MaskProps, {}> {

  static defaultProps = {
    prefixCls: 'za-mask',
    visible: false,
    type: 'normal',
  };

  render(): any {
    const { prefixCls, className, visible, type, onClose, ...others } = this.props;
    const markCls = classnames(`${prefixCls}`, className, type);

    return visible && <div className={markCls} onClick={onClose} {...others} />;
  }
}
