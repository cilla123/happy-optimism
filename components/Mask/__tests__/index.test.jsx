import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Mask from '../index';

describe('Mask', () => {
  it('renders correctly', () => {
    const wrapper = render(<Mask visible />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
