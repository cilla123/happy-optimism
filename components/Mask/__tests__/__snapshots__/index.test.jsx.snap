// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mask renders correctly 1`] = `
<div
  class="za-mask normal"
/>
`;
